scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/util/AllObjects2BytesConverterHandler.java

```diff
@@ -58,30 +62,46 @@ private AllObjects2BytesConverterHandler() {
 
     public static Map<String, Serializable> convertAllBytes2Objects(final Map<String, byte[]> target,
             final ClassLoader cl) {
-        return convert("Deserialization of variables", createDeserializeCollable(target, cl));
+        long totalBytes = target != null ? target.values()
+                                                 .stream()
+                                                 .mapToInt(value -> value != null ? value.length : 0)
+                                                 .sum()
+                                         : 0;
+        return convert("Deserialization of variables", createDeserializeCollable(target, cl), totalBytes);
     }
 
     public static Map<String, byte[]> convertAllObjects2Bytes(final Map<String, Serializable> variableMap) {
-        return convert("Serialization of variables", createSerializableCallable(variableMap));
+        return convert("Serialization of variables", createSerializableCallable(variableMap), Long.MAX_VALUE);
     }
 
     public static byte[] convertObject2Byte(final String key, final Serializable value) {
-        return convertSingle("Serialization of single value", createSerializeSingleValueCollable(key, value));
+        return convertSingle("Serialization of single value",
+                             createSerializeSingleValueCollable(key, value),
+                             Long.MAX_VALUE);
     }
 
     public static Serializable convertByte2Object(final byte[] value) {
-        return convertSingle("Deserialization of single value", createDeserializeSingleValueCollable(value));
+        return convertSingle("Deserialization of single value",
+                             createDeserializeSingleValueCollable(value),
+                             value != null ? value.length : 0);
     }
 
-    private static <K, V> Map<K, V> convert(String action, Callable<Map<K, V>> callable) {
+    private static <K, V> Map<K, V> convert(String action, Callable<Map<K, V>> callable, long totalSize) {
 
         Map<K, V> resultMap = new HashMap<>();
 
         ExecutorService executor = Executors.newSingleThreadExecutor();
         Future<Map<K, V>> future = executor.submit(callable);
 
         try {
-            resultMap = future.get(secondsToWait, TimeUnit.SECONDS);
+            if (totalSize < deserializationThreshold) {
+                // If the amount of bytes to deserialize is small, the deserialization should not exceed a timeout,
+                // otherwise it means a deadlock occurs (most likely due to jython types)
+                // timeout is disabled for serialization
+                resultMap = future.get(secondsToWait, TimeUnit.SECONDS);
+            } else {
+                resultMap = future.get();
+            }
```

Comment:
I don't understand why the deserialization is still executed when the totalSize is greater than deserializationThreshold

Reply:
we don't want to prevent deserialization for big size data. In that case, we don't use a timeout as we know the deserialization can take some time. The timeout is used for small size data, where it's not normal to exceed a default timeout.

Reply:
what if the pb of deadlock occurs for a big size data? how to detect that?

Reply:
indeed, we cannot detect that